plugins/inputs/marklogic/marklogic.go

```diff
@@ -0,0 +1,233 @@
+package marklogic
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	dac "github.com/xinsnake/go-http-digest-auth-client"
+)
+
+type Marklogic struct {
+	Hosts          []string `toml:"hosts"`
+	DigestUsername string   `toml:"digest_username"`
```

Comment:
This could just be `Username` and `Password` to be more uniform with the rest of telegraf's plugins.